src/main/java/io/kuzzle/sdk/core/KuzzleDocument.java

```diff
@@ -307,23 +300,30 @@ public KuzzleDocument publish() {
   /**
    * Sets content
    *
-   * @param data    the data
+   * @param content    the data
    * @param replace the replace
    * @return content content
    */
-  public KuzzleDocument setContent(final JSONObject data, final boolean replace) {
-    try {
-      if (replace) {
-        put("body", data);
-      } else {
-        for (Iterator iterator = data.keys(); iterator.hasNext(); ) {
-          String key = (String) iterator.next();
-          getJSONObject("body").put(key, data.get(key));
-        }
+  public KuzzleDocument setContent(final JSONObject content, final boolean replace) throws JSONException {
+    if (replace) {
+      if (content != null) {
+        this.content = new JSONObject(content.toString());
```

Comment:
Never mind, only the javax.json.JsonObject is immutable and not the org.json.JsonObject